pydantic/fields.py

```diff
@@ -279,6 +285,12 @@ def _populate_validators(self) -> None:
             self.whole_pre_validators = self._prep_vals(v.func for v in class_validators_ if v.whole and v.pre)
             self.whole_post_validators = self._prep_vals(v.func for v in class_validators_ if v.whole and not v.pre)
 
+        if self.sub_fields and self.schema is not None and self.schema.const:
```

Comment:
can you add a comment explaining what this is doing please.

Reply:
done